### PR TITLE
update requests stubs with various missing methods

### PR DESCRIPTION
**Pull request by vbarbaresi**
I used the latest `requests==2.24.0` as reference and ran stubtest. I fixed and updated a few things

### CODE REVIEW

File: third_party/2and3/requests/utils.pyi [hunk 1]
```diff
@@ -41,9 +41,9 @@ def dotted_netmask(mask): ...
 def is_ipv4_address(string_ip): ...
 def is_valid_cidr(string_network): ...
 def set_environ(env_name, value): ...
-def should_bypass_proxies(url): ...
-def get_environ_proxies(url): ...
-def select_proxy(url, proxies): ...
+def should_bypass_proxies(url, no_proxy: Iterable[AnyStr]) -> bool: ...
+def get_environ_proxies(url, no_proxy: Iterable[AnyStr] = ...) -> Dict[Any, Any]: ...
```
Comment (srittau) [inline third_party/2and3/requests/utils.pyi:45]: There is an explicit `None` check for `no_proxy`.

File: third_party/2and3/requests/utils.pyi [hunk 2]
```diff
@@ -41,9 +41,9 @@ def dotted_netmask(mask): ...
...
+def select_proxy(url, proxies: Dict[Any, Any]): ...
```
Comment (srittau) [inline third_party/2and3/requests/utils.pyi:46]: There is this line in the implementation: `proxies = proxies or {}`, so `None` is accepted. Also `Mapping` should be sufficient:

```suggestion
def select_proxy(url: Text, proxies: Optional[Mapping[Any, Any]]): ...
```

File: third_party/2and3/requests/utils.pyi [hunk 3]
```diff
@@ -53,3 +53,4 @@ def get_auth_from_url(url): ...
 def to_native_string(string, encoding=...): ...
 def urldefragauth(url): ...
 def rewind_body(prepared_request): ...
+def check_header_validity(header: Tuple[AnyStr, ...]) -> None: ...
```
Comment (srittau) [inline third_party/2and3/requests/utils.pyi:56]: As `header` is a two-element tuple, not an arbitrary sized one:

```suggestion
def check_header_validity(header: Tuple[AnyStr, AnyStr]) -> None: ...
```

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks, remarks below.

**Comment by srittau** [comment]
We now had a merge conflict with master, which I resolved.

**Comment by vbarbaresi** [comment]
Thanks for the thorough review and the conflict resolution!
I checked and applied all your suggestions 👌

**Review by srittau — APPROVED** [review]

**Pull request merged**